Error on Field.limit_choices_to with callable

# Bug report

<!--
Hi, thanks for submitting a bug. We appreciate that.

But, we will need some information about what's wrong to help you.
-->

## What's wrong

I get a mypy error when I use a callable on a `ForeignKey.limit_choices_to` argument:

```
error: Argument "limit_choices_to" to "ForeignKey" has incompatible type "Callable[[], dict[str, set[Any]]]"; expected "Q | dict[str, Any] | _ChoicesCallable | None"  [arg-type]
```

My code looks like this:

```python
def limit_to_procedures() -> dict[str, set[Any]]:
    return {"pk__in": {content_type.pk for content_type in ContentType.objects.get_for_models(*AbstractModel.__subclasses__()).values()}} 

def Foo(Model):
  content_type = ForeignKey(ContentType, limit_choices_to=limit_to_procedures, on_delete=CASCADE)
```
<!--
Describe what is not working.

Please, attach a traceback.
We would also appreciate a failing test case.
That is EXTREMELY helpful!
-->

## How is that should be

<!-- Describe how it should work. -->
No error should be thrown.

## System information

- OS: Linux
- `python` version: 3.12.7
- `django` version: 5.1.1
- `mypy` version: 1.11.2
- `django-stubs` version: 5.1.0
- `django-stubs-ext` version: 5.1.0

--- a/django-stubs/db/models/fields/__init__.pyi
+++ b/django-stubs/db/models/fields/__init__.pyi
@@ -26,6 +26,7 @@ BLANK_CHOICE_DASH: list[tuple[str, str]]
 
 _ChoicesList: TypeAlias = Sequence[_Choice] | Sequence[_ChoiceNamedGroup]
 _LimitChoicesTo: TypeAlias = Q | dict[str, Any]
+_LimitChoicesToCallable: TypeAlias = Callable[[], _LimitChoicesTo]
 
 _F = TypeVar("_F", bound=Field, covariant=True)
 
@@ -39,7 +40,7 @@ class _FieldDescriptor(Protocol[_F]):
     @property
     def field(self) -> _F: ...
 
-_AllLimitChoicesTo: TypeAlias = _LimitChoicesTo | _ChoicesCallable  # noqa: PYI047
+_AllLimitChoicesTo: TypeAlias = _LimitChoicesTo | _LimitChoicesToCallable | _ChoicesCallable  # noqa: PYI047
 _ErrorMessagesMapping: TypeAlias = Mapping[str, _StrOrPromise]
 _ErrorMessagesDict: TypeAlias = dict[str, _StrOrPromise]
 

